grails-app/controllers/org/pih/warehouse/inventory/StockMovementController.groovy

```diff
@@ -244,6 +245,49 @@ class StockMovementController {
         redirect(action: "list", params: params << ["flash": flash as JSON])
     }
 
+    def updateStatus = {
+        StockMovement stockMovement = stockMovementService.getStockMovement(params.id)
+
+        try {
+            stockMovementService.transitionStockMovement(stockMovement, params as JSONObject)
+
+            switch(stockMovement.requisition?.status) {
+                case RequisitionStatus.APPROVED:
+                    flash.message = g.message(
+                            code: "request.approved.message.label",
+                            default: "You have successfully approved the request {0}",
+                            args: [stockMovement.id]
+                    )
+                    break
+                case RequisitionStatus.REJECTED:
+                    flash.message = g.message(
+                            code: "request.rejected.message.label",
+                            default: "You have successfully rejected the request {0}",
+                            args: [stockMovement.id]
+                    )
+                    break
+                default:
+                    break
+            }
```

Comment:
Maybe we can shorten it to, but if you don't like this solution, let's leave it as is
```
if (stockMovement.requisition?.status in [RequisitionStatus.APPROVED, RequisitionStatus.REJECTED]) {
   String status = stockMovement.requisition?.status?.toLowerCase()   
   flash.message = g.message(
                            code: "request.${status}.message.label",
                            default: "You have successfully ${status} the request {0}",
                            args: [stockMovement.id]
                    )
}
```